pkg/proxy/ipvs/proxier.go

```diff
@@ -516,15 +525,53 @@ func (handle *LinuxKernelHandler) GetModules() ([]string, error) {
 	}
 
 	// Find out loaded kernel modules
-	out, err := handle.executor.Command("cut", "-f1", "-d", " ", "/proc/modules").CombinedOutput()
+	modulesFile, err := os.Open("/proc/modules")
 	if err != nil {
 		return nil, err
 	}
 
-	mods := strings.Split(string(out), "\n")
+	mods, err := getFirstWordOfEachLine(modulesFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find loaded kernel modules: %v", err)
+	}
+
 	return append(mods, bmods...), nil
 }
 
+// getFirstWordOfEachLine reads all the content from r into memory and return a
+// slice which consists of the first word from each line.
+func getFirstWordOfEachLine(r io.Reader) ([]string, error) {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	mods := strings.Split(string(b), "\n")
+	for i := range mods {
+		fields := strings.Fields(mods[i])
+		if len(fields) > 0 {
+			mods[i] = fields[0]
+		}
+	}
+	return mods, nil
+}
+
+// GetKernelVersion returns currently running kernel version.
+func (handle *LinuxKernelHandler) GetKernelVersion() (string, error) {
+	kernelVersionFile := "/proc/sys/kernel/osrelease"
+	fileContent, err := ioutil.ReadFile(kernelVersionFile)
+	if err != nil {
+		return "", fmt.Errorf("error reading osrelease file %q: %v", kernelVersionFile, err)
+	}
+
+	fields := strings.Fields(string(fileContent))
```

Comment:
On second thought, this can just be strings.TrimSpace - same as before. I think strings.Fields is only required for the modules